### PR TITLE
applications: nrf5340_audio: Handle pref_pres_dly of 0

### PR DESCRIPTION
**Pull request by alexsven**
- Some devices set their preferred pres delay to 0, in those cases we need to set pres delay to common minimum

### CODE REVIEW

**Review by rick1082 — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 1]
```diff
@@ -246,7 +246,8 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
 		return 0;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN)) {
+	/* If preferred delay from unicast_server is 0 it is up to unicast_server to select delay */
+	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN) && pref_dly_min != 0) {
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:250]: We need a check that the end pres_dly_us is in valid range.

**Review by koffes — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c (hunk 1 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c]: Discussed offline and it is not needed. The range of the gateway is not necessary to take into account.

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 2]
```diff
@@ -247,38 +250,29 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN)) {
-		*pres_dly_us = pref_dly_min;
+		if (pref_dly_min == 0) {
+			*pres_dly_us = pres_dly_min;
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:254]: Nice. Suggest a comment to explain why this is done.

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 3]
```diff
@@ -247,38 +250,29 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
...
+		} else {
+			/* Preferred min is 0, so we set min supported */
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:256]: Should this comment be on the case above?

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 4]
```diff
@@ -247,38 +250,29 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
...
-			if (pres_dly_max < CONFIG_BT_AUDIO_PRESENTATION_DELAY_US) {
-				*pres_dly_us = pres_dly_max;
-
-				LOG_WRN("Selecting maximum common delay: %d us ", pres_dly_max);
-			} else {
-				*pres_dly_us = pres_dly_min;
-
-				LOG_WRN("Selecting minimum common delay: %d us ", pres_dly_min);
-			}
+			/* Preferred min is 0, so we set max supported */
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:267]:
```suggestion
			/* Preferred max is 0, so we set max supported */
```

**Review by koffes — CHANGES_REQUESTED** [review]

**Review by gWacey — APPROVED** [review]

**Review by koffes — APPROVED** [review]

**Pull request merged**